src/graphapi/graphapi_tensor.cpp

```diff
@@ -257,7 +258,10 @@ void BackendTensorDescriptor::getAttribute(miopenBackendAttributeName_t attribut
             const auto& dimensions = mDescriptor.getDimensions();
             *elementCount          = dimensions.size();
             std::copy_n(dimensions.begin(),
-                        std::min(*elementCount, requestedElementCount),
+                        // WORKAROUND: building on Windows is failing due to conflicting definitions
+                        // of std::min() between the MSVC standard library and HIP Clang wrappers.
+                        *elementCount < requestedElementCount ? *elementCount :
+                                                                requestedElementCount,
```

Comment:
It looks bad. And suspicious.
It is the first encounter of the problem, while `std::min` is widely used.

By the way, why does it use cuda wrapper?

Could you double check that the problem is in the code and not in the configuration?

Reply:
It is more than certain that the problem is in the compiler and its configuration, not the MIOpen code. However, we will not get the compiler fixed in a short time. At the same time, we need to continue working. With this issue present, the building on Windows is broken.

Reply:
I just checked on Linux, and the `<algorithm>` header file is taken from `/opt/rocm-6.0.2/lib/llvm/lib/clang/17.0.0/include/cuda_wrappers/`. It might be related to the MIOpen build configuration. MSVC is hitting a conflict, but GNU C++ is not.

Reply:
Absence of calls of a template function should not fix the redefinition of the function in c++17.

Two new folders we added in PR [#2751](https://github.com/ROCm/MIOpen/pull/2751)
- src/graphapi/
- src/include/miopen/graphapi/

Might it affect the build system?